docs/proposals/19-monitoring.md

```diff
@@ -0,0 +1,373 @@
+centralize the configuration, it is possible to improve the developer experience
+and improve the general stability of the monitoring. This can be done by
+introducing the [prometheus-operator]. This operator will make it easier for
+monitoring configuration to be discovered and picked up with the use of the
+[Custom Resources][prom-crds] provided by the prometheus-operator. These
+resources can also be directly referenced in Go and be deployed by their
+respective components, instead of creating `.yaml` files in Go, or templating
+charts. With the addition of the prometheus-operator it should also be easier to
+add new features, such as Thanos.
+
+## Motivation
+
+Simplify monitoring changes and extensions with the use of the
+[prometheus-operator]. The current extension contract is described
+[here][extension-contract]. This document aims to define a new contract.
+
+Make it easier to add new monitoring features and make new changes. For example
+using the [prometheus-operator]
+
+The prometheus-operator handles validation of monitoring configuration. It will
+be more difficult to give prometheus invalid config.
+
+### Goals
+
+- The [prometheus-operator] and the [Custom Resource
+  Definitions][prom-crds]
+  will be deployed in each seed cluster.
+
+- Gardener and its extensions will deploy [Custom Resources][prom-crds] provided
+  by the prometheus-operator to create prometheis and configuration for control
+  planes and extensions. This configuration should create a monitoring stack
+  that mirrors the current implmentation.
+
+- Leverage community knowledge by using [kube-prometheus] and
+  [kubernetes-mixins]. Reuse existing dashboards and monitoring configuration
+  for Kubernetes components whenever possible.
+
+- Add a [sidecar][grafana-sidecar] to Grafana to discover and provision dashboards.
+
+- Migrate from the current monitoring stack to the prometheus-operator.
+
+### Non-Goals
+
+- Changes to the logging stack.
+
+- Feature parity between the current solution and the one proposed in this GEP.
+
+## Proposal
+
+Today, Gardener provides monitoring for shoot clusters (i.e. System Components
+and the control plane) and for the seed cluster. The proposal is to migrate
+these monitoring stacks to use the [prometheus-operator]. The proposal is lined
+out in the following steps:
+
+0. Define how will Gardener generate the Prometheus objects.
+    1. Option 1: Use the [API] provided by the [prometheus-operator] and create Go
+       structs.
+    2. Option 2: Template `yaml` files. The `yaml` should not be inline in Go,
+       but rather exist on its own. For some of the monitoring configuration it
+       is nice to look at the `yaml` representation rather than the Go structs.
+
+1. Deploy the [prometheus-operator] and its CRDs. It is not clear what the best
+   way to deploy this is yet. It could be similar to how the Vertical Pod
+   Autoscaler is [deployed][vpa]. The [prometheus-operator] can also be a
```

Comment:
One concern I had is that all of the [CRDs](https://github.com/prometheus-operator/prometheus-operator/blob/main/bundle.yaml) are about 1.5MB. This would exceed the maximum size of secrets/configmaps.

Reply:
We don't need to put all CRDs into one secret, so that should be fine.
Additionally, we could check which CRDs we need and want to use :)